apps/src/templates/sectionProgress/standards/StandardsIntroDialog.jsx

```diff
@@ -0,0 +1,84 @@
+import PropTypes from 'prop-types';
+import React, {Component} from 'react';
+import Portal from 'react-portal';
+import i18n from '@cdo/locale';
+import Dialog, {Title, Body} from '@cdo/apps/templates/Dialog';
+import color from '@cdo/apps/util/color';
+import SafeMarkdown from '@cdo/apps/templates/SafeMarkdown';
+
+const styles = {
+  description: {
+    color: color.dark_charcoal
+  },
+  boldText: {
+    fontFamily: '"Gotham 7r", sans-serif'
+  }
+};
+
+export class StandardsIntroDialog extends Component {
+  static propTypes = {
+    isOpen: PropTypes.bool.isRequired,
+    handleConfirm: PropTypes.func.isRequired,
+    handleCancel: PropTypes.func.isRequired
+  };
+
+  render() {
+    return (
+      <Dialog
+        isOpen={this.props.isOpen}
+        confirmText={i18n.gotIt()}
+        onConfirm={this.props.handleConfirm}
+      >
+        <Title>{i18n.progressOnCSTAStandards()}</Title>
+        <Body>
+          <div style={styles.description}>
+            <p>
+              {i18n.progressOnCSTAStandardsDescription()}{' '}
+              <a
+                href="https://www.csteachers.org/page/standards"
+                target="_blank"
+                style={styles.boldText}
+              >
+                {i18n.CSTAStandards()}
+              </a>
+            </p>
+          </div>
+          <div style={styles.description}>
+            <p>{i18n.useToView()}</p>
+            <ul>
+              <li>
+                <SafeMarkdown markdown={i18n.useToViewList1()} />
+              </li>
+              <li>
+                <SafeMarkdown markdown={i18n.useToViewList2()} />
+              </li>
+              <li>
+                <SafeMarkdown markdown={i18n.useToViewList3()} />
+              </li>
+            </ul>
+          </div>
+          <div style={styles.description}>
+            <p>
+              <SafeMarkdown markdown={i18n.standardsReminder()} />
+            </p>
+          </div>
+        </Body>
+      </Dialog>
+    );
+  }
+}
+
+// Our default export is actually a wrapper around our dialog that renders it
+// through a Portal component so it sits at the end of the DOM instead of
+// inside whatever component called for it - but this is lousy for testing,
+// so we mostly export and test the inner dialog component.
+export default class StandardsIntroDialogPortal extends Component {
```

Comment:
I'm confused about why we need this part. Is this to avoid exporting an unconnected component? Why use the `Dialog` component rather than `BaseDialog`?

Reply:
I'm new to this pattern - what is the benefit of moving the dialog to the end of the DOM?

Reply:
I also do not understand this motivation for this pattern. It looks like it is copied from `ConfirmEnableMakerDialog`. @islemaster , can you share any insight as to why this can't just be a dialog colocated in the DOM with whichever component renders it?

Reply:
Ya, I just used ConfirmEnableMakerDialog as an example here. If there is a better example please do share.  @Erin007 what is the difference between Dialog and BaseDialog?

Reply:
I just updated it to use BaseDialog instead

Reply:
And removed the Portal thing since I found the other Dialog examples which didn't have that.

Reply:
Thanks for the follow up, Dani!  It's confusing to have so many different examples and Dialog variations floating around.  My inclination toward BaseDialog is that Dialog still used LegacyButton.  Lots of opportunity to consolidate and clean up 🙃